pkg/pillar/types/assignableadapters.go

```diff
@@ -67,6 +79,82 @@ type IoBundle struct {
 	IsPort    bool // Whole or part of the bundle is a zedrouter port
 }
 
+// HasAdapterChanged - We store each Physical Adapter using the IoBundle object.
+// Compares IoBundle with Physical adapter and returs if they are the Same
+// or the Physical Adapter has changed.
+func (ib IoBundle) HasAdapterChanged(phyAdapter PhysicalIOAdapter) bool {
+	if IoType(phyAdapter.Ptype) != ib.Type {
+		log.Infof("Type changed from %d to %d", ib.Type, phyAdapter.Ptype)
+		return true
+	}
+	if phyAdapter.Phylabel != ib.Name {
+		log.Infof("Name changed from %s to %s", ib.Name, phyAdapter.Phylabel)
+		return true
+	}
+	if phyAdapter.Phyaddr.PciLong != ib.PciLong {
+		log.Infof("PciLong changed from %s to %s",
+			ib.PciLong, phyAdapter.Phyaddr.PciLong)
+		return true
+	}
+	if phyAdapter.Phyaddr.Ifname != ib.Ifname {
+		log.Infof("Ifname changed from %s to %s",
+			ib.Ifname, phyAdapter.Phyaddr.Ifname)
+		return true
+	}
+	if phyAdapter.Phyaddr.Serial != ib.Serial {
+		log.Infof("Serial changed from %s to %s",
+			ib.Serial, phyAdapter.Phyaddr.Serial)
+		return true
+	}
+	if phyAdapter.Phyaddr.Irq != ib.Irq {
+		log.Infof("Irq changed from %s to %s", ib.Irq, phyAdapter.Phyaddr.Irq)
+		return true
+	}
+	if phyAdapter.Phyaddr.Ioports != ib.Ioports {
+		log.Infof("Ioports changed from %s to %s",
+			ib.Ioports, phyAdapter.Phyaddr.Ioports)
+		return true
+	}
+	if phyAdapter.Logicallabel != ib.Logicallabel {
+		log.Infof("Logicallabel changed from %s to %s",
+			ib.Logicallabel, phyAdapter.Logicallabel)
+		return true
+	}
+	if phyAdapter.Assigngrp != ib.AssignmentGroup {
+		log.Infof("Ifname changed from %s to %s",
+			ib.AssignmentGroup, phyAdapter.Assigngrp)
+		return true
+	}
+	if phyAdapter.Usage != ib.Usage {
+		log.Infof("Usage changed from %d to %d", ib.Usage, phyAdapter.Usage)
+		return true
+	}
+	if phyAdapter.UsagePolicy.FreeUplink != ib.FreeUplink {
```

Comment:
Can phyAdapter.UsagePolicy be nil?

Reply:
no.. phyAdapter is not the same structure as used in ProtoBuf.. This is a local structure. Here, UsagePolicy us an embedded field.. not a ptr